autoscoper/src/ui/GLTimeline.cpp

```diff
@@ -192,7 +192,23 @@ void GLTimeline::mouseMoveEvent(QMouseEvent *e){
                 if (type == NODE) {
                     //node.set_x(node.get_x()+dx); // Prevent x from begin
                     //modified
-                    curve.set_value(it,curve.value(it)+dy);
+                  if (curve.type != IKeyCurve::QUAT_CURVE) {
```

Comment:
Since you are already relying on `dynamic_cast` to differentiate `float`-based vs `Quaternion`-based curve, consider removing the explicit check for the type ...